### DISCUSSION TITLE
Setting environment variable using REST API

### DISCUSSION THREAD

**Discussion by Mofiqul** [vercel/community#1266] [Help] [answered]
### Summary

Hi, I am using REST API to deploy our multi tenant website. The problem is that I need to set environment variable per deployment. In documentation, I found out that send environment variable is deprecated. And I tried to send it did not work either. Can someone please help me out. 

### Additional information

I am using https://vercel.com/docs/rest-api#endpoints/deployments/create-a-new-deployment end point

**Comment by thmsmtylr** [comment, marked as answer]
@Mofiqul I ran into the same issue I haven't verified this works yet but I think you can now use: https://vercel.com/docs/rest-api#endpoints/projects/create-one-or-more-environment-variables. Hopefully that helps :)

**Comment by Mofiqul** [comment]
@amyegan 
What is structure of body parameters? 
I am sending this 
```json
[
  {
    "key": "NEXT_PUBLIC_ORG_ID",
    "value": "value here",
    "type": "plain",
    "target": "production"
  },
  {
    "key": "NEXT_PUBLIC_MEMBERSHIP_ROLE",
    "value": "SUBSCRIBER",
    "type": "plain",
    "target": "production"
  },
  {
    "key": "NEXT_PUBLIC_BASE_URL",
    "value": "value here",
    "type": "plain",
    "target": "production"
  }
]
```
Getting the following error 
```json
{
  "error": {
    "code": "bad_request",
    "message": "Invalid request: should be object."
  }
}
```

**Reply by thmsmtylr** [reply]
The most obvious issue is that the `target` key value should be an Array: `["target": "production"]`. This is what I'm using successfully:

```
[{
    type: "encrypted",
    target: ["production", "preview", "development"],
    key: "MY_KEY",
    value: "my key value",
 }]
  ```

But you're getting an object error so I'm assuming `type`:`plain` may require an object.

If you hit the API using something like [Insomnia](https://insomnia.rest/download) with your above array you'll get a more descriptive error.

**Reply by amyegan** [reply]
Thomas has the right idea. You'll need to make the `target` an array, even if it is only one value. 

Thanks for calling attention to this flaw in our documentation. I know the team is already working on some updates for the API docs, but I'll point this out to make sure it doesn't get missed. 🙇

**Reply by itsjavi** [reply]
The docs are misleading when it comes to the HTTP method. It should be in capitals (e.g. `PATCH` instead of `patch`), at least for the endpoint to edit an env var https://vercel.com/docs/rest-api/endpoints#edit-an-environment-variable

Otherwise I get "BAD REQUEST" as plain-text response all the time. It was hard to figure out what was going on.


@amyegan

**Reply by stuburger** [reply]
Wow This lower case `"method": "patch"` in the docs (https://vercel.com/docs/rest-api/endpoints#edit-an-environment-variable) had me wondering what I was doing wrong for hours. Thanks finally saving me @itsjavi  🤗
